example/src/minimal_publisher.cpp

```diff
@@ -26,61 +26,48 @@
-    timer_ = this->create_wall_timer(
-        500ms, std::bind(&MinimalPublisher::timer_callback, this));
-    param_listener = std::make_shared<admittance_controller::ParamListener>(
-        get_node_parameters_interface());
-    params_ = param_listener->get_params();
+namespace admittance_controller {
 
-    admittance_controller::StackParams s_params =
-        param_listener->get_stack_params();
+MinimalPublisher::MinimalPublisher(const rclcpp::NodeOptions& options)
+    : Node("admittance_controller", options) {
+  timer_ = create_wall_timer(
+      500ms, std::bind(&MinimalPublisher::timer_callback, this));
+  param_listener =
+      std::make_shared<ParamListener>(get_node_parameters_interface());
+  params_ = param_listener->get_params();
 
-    RCLCPP_INFO(this->get_logger(), "Initial control frame parameter is: '%s'",
+  StackParams s_params = param_listener->get_stack_params();
+
+  RCLCPP_INFO(get_logger(), "Initial control frame parameter is: '%s'",
+              params_.control.frame.id.c_str());
+  RCLCPP_INFO(get_logger(), "fixed string is: '%s'",
+              std::string{params_.fixed_string}.c_str());
+  const tcb::span<double> fixed_array = params_.fixed_array;
+  for (auto d : fixed_array) {
+    RCLCPP_INFO(get_logger(), "value: '%s'", std::to_string(d).c_str());
+  }
+}
+
+void MinimalPublisher::timer_callback() {
+  if (param_listener->is_old(params_)) {
+    param_listener->refresh_dynamic_parameters();
+    params_ = param_listener->get_params();
+    RCLCPP_INFO(get_logger(), "New control frame parameter is: '%s'",
                 params_.control.frame.id.c_str());
-    RCLCPP_INFO(this->get_logger(), "fixed string is: '%s'",
+    RCLCPP_INFO(get_logger(), "fixed string is: '%s'",
                 std::string{params_.fixed_string}.c_str());
     const tcb::span<double> fixed_array = params_.fixed_array;
     for (auto d : fixed_array) {
-      RCLCPP_INFO(this->get_logger(), "value: '%s'", std::to_string(d).c_str());
+      RCLCPP_INFO(get_logger(), "value: '%s'", std::to_string(d).c_str());
     }
   }
-
- private:
-  void timer_callback() {
-    if (param_listener->is_old(params_)) {
-      param_listener->refresh_dynamic_parameters();
-      params_ = param_listener->get_params();
-      RCLCPP_INFO(this->get_logger(), "New control frame parameter is: '%s'",
-                  params_.control.frame.id.c_str());
-      RCLCPP_INFO(this->get_logger(), "fixed string is: '%s'",
-                  std::string{params_.fixed_string}.c_str());
-      const tcb::span<double> fixed_array = params_.fixed_array;
-      for (auto d : fixed_array) {
-        RCLCPP_INFO(this->get_logger(), "value: '%s'",
-                    std::to_string(d).c_str());
-      }
-    }
-  }
-
-  rclcpp::TimerBase::SharedPtr timer_;
-  std::shared_ptr<admittance_controller::ParamListener> param_listener;
-  admittance_controller::Params params_;
-};
-
-int main(int numArgs, const char** args) {
```

Comment:
I understand that for the Compoennts one doesn't need `main`. Nevertheless, would it make sense to have both examples, Component Node and "normal" Node?

Reply:
When you create a component node it also creates an executable. You can see from the cmake and the readme of the example that there is still an executable and this file is now redundant.